packages/tracing/src/hubextensions.ts

```diff
@@ -183,12 +189,19 @@ export function startIdleTransaction(
   onScope?: boolean,
   customSamplingContext?: CustomSamplingContext,
 ): IdleTransaction {
-  const transaction = new IdleTransaction(transactionContext, hub, idleTimeout, onScope);
-  return sample(hub, transaction, {
+  const client = hub.getClient();
+  const options = (client && client.getOptions()) || {};
+
+  let transaction = new IdleTransaction(transactionContext, hub, idleTimeout, onScope);
+  transaction = sample(transaction, options, {
     parentSampled: transactionContext.parentSampled,
     transactionContext,
     ...customSamplingContext,
   });
+  if (transaction.sampled) {
+    transaction.initSpanRecorder(options._experiments?.maxSpans as number);
+  }
+  return transaction;
```

Comment:
Not cool that we have to repeat ourselves in `_startTransaction` and `startIdleTransaction`, but letting it be for now as we may soon move away from idle transactions anyway.